core/src/consensus.rs

```diff
@@ -130,19 +130,57 @@ pub const HARD_FORK_INTERVAL: u64 = YEAR_HEIGHT / 2;
 /// Check whether the block version is valid at a given height, implements
 /// 6 months interval scheduled hard forks for the first 2 years.
 pub fn valid_header_version(height: u64, version: HeaderVersion) -> bool {
-	// uncomment below as we go from hard fork to hard fork
-	if height < HARD_FORK_INTERVAL {
-		version == HeaderVersion::default()
-	/* } else if height < 2 * HARD_FORK_INTERVAL {
-		version == 2
-	} else if height < 3 * HARD_FORK_INTERVAL {
-		version == 3
-	} else if height < 4 * HARD_FORK_INTERVAL {
-		version == 4
-	} else if height >= 5 * HARD_FORK_INTERVAL {
-		version > 4 */
-	} else {
-		false
+	let chain_type = global::CHAIN_TYPE.read().clone();
+	match chain_type {
+		global::ChainTypes::Mainnet => {
+			// uncomment below as we go from hard fork to hard fork
+			if height < HARD_FORK_INTERVAL {
+				version == HeaderVersion::default()
+			} else if height < 2 * HARD_FORK_INTERVAL {
+				version == HeaderVersion::new(2)
+			/*} else if height < 3 * HARD_FORK_INTERVAL {
+				version == HeaderVersion::new(3)
+			} else if height < 4 * HARD_FORK_INTERVAL {
+				version == HeaderVersion::new(4)
+			} else if height >= 5 * HARD_FORK_INTERVAL {
+				version > HeaderVersion::new(4) */
+			} else {
+				false
+			}
+		}
+		global::ChainTypes::Floonet => {
+			// uncomment below as we go from hard fork to hard fork
+			if height < HARD_FORK_INTERVAL - 28 * DAY_HEIGHT {
```

Comment:
Maybe its better to have a hardcoded height here and set the next one to that height plus `HARD_FORK_INTERVAL`?

Reply:
Why would we want to use `HARD_FORK_INTERVAL` at all when the exact heights are static and known?  The fewer calculations performed in consensus code the safer and predictable it all feels.

Reply:
The exact height for the second hard fork on Floonet is not yet known. It is unlikely that blocks will keep coming in at a 1 minute interval for 6 months straight, so i see it as setting an upperbound now, then when we get close to the next hardfork we can adjust it to an explicit height.

Reply:
👍 @jaspervdm on setting an exact block height. Floonet is currently at bock 165,019 which means choosing `HARD_FORK_INTERVAL - 28 * DAY_HEIGHT` does not make any sense for a fork around June 20.

Reply:
Added `FLOONET_FIRST_HARD_FORK` at height 185,200.